earthfile2llb/interpreter.go

```diff
@@ -1657,8 +1657,11 @@ func (i *Interpreter) expandArgs(ctx context.Context, word string, keepPlusEscap
 		WithShell:   true,
 	}
 
-	ret, err := i.converter.ExpandArgs(ctx, runOpts, escapeSlashPlus(word))
+	ret, err := i.converter.ExpandArgs(ctx, runOpts, escapeSlashPlus(word), !async)
 	if err != nil {
+		if errors.Is(err, errShellOutNotPermitted) {
```

Comment:
Slightly more future-proof...

```suggestion
		if async && errors.Is(err, errShellOutNotPermitted) {
```